File: backend/_dev_scripts/check_types.js

const { Pool } = require('pg');
require('dotenv').config();

const pool = new Pool({
    user: process.env.DB_USER,
    host: process.env.DB_HOST,
    database: process.env.DB_NAME,
    password: process.env.DB_PASSWORD,
    port: process.env.DB_PORT,
});

async function checkTypes() {
    try {
        const res = await pool.query(`
            SELECT column_name, data_type 
            FROM information_schema.columns 
            WHERE table_name = 'pedidos' 
            AND column_name IN ('ped_cliind', 'ped_nffat');
        `);
        console.log(res.rows);
    } catch (err) {
        console.error(err);
    } finally {
        pool.end();
    }
}

checkTypes();

File: bi-engine/check_types.py
from services.database_native import db

def check_types():
    try:
        res = db.execute_query("SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'itens_ped' AND column_name LIKE 'ite_%%'")
        print("--- itens_ped ---")
        for r in res: print(r)
        
        res = db.execute_query("SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'cad_prod' AND column_name LIKE 'pro_%%'")
        print("\n--- cad_prod ---")
        for r in res: print(r)
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    check_types()
